tmt.spec

```diff
@@ -1,250 +1,109 @@
-%if 0%{?rhel} == 8
-BuildRequires: python%{python3_pkgversion}-typing-extensions
-BuildRequires: python%{python3_pkgversion}-dataclasses
-BuildRequires: python%{python3_pkgversion}-importlib-metadata
-%endif
-# Required for tests
-BuildRequires: rsync
-%{?python_provide:%python_provide python%{python3_pkgversion}-%{name}}
+BuildArch:      noarch
+BuildRequires:  python3-devel
 
-%description -n python%{python3_pkgversion}-%{name}
+%global _description %{expand:
 The tmt Python module and command line tool implement the test
-metadata specification (L1 and L2) and allows easy test execution.
-This package contains the Python 3 module.
-
-%package provision-container
-Summary: Container provisioner for the Test Management Tool
-Obsoletes: tmt-container < 0.17
-Requires: tmt == %{version}-%{release}
-Requires: podman
-Requires: (ansible or ansible-collection-containers-podman)
-
-%description provision-container
-Dependencies required to run tests in a container environment.
-
-%package provision-virtual
-Summary: Virtual machine provisioner for the Test Management Tool
-Obsoletes: tmt-testcloud < 0.17
-Requires: tmt == %{version}-%{release}
-Requires: python%{python3_pkgversion}-testcloud >= 0.9.2
-Requires: libvirt-daemon-config-network
-Requires: openssh-clients
-Requires: (ansible or ansible-core)
+metadata specification (L1 and L2) and allows easy test execution.}
+
+%description %_description
+
+Summary:        %{summary}
+Requires:       git-core rsync sshpass
+
+Obsoletes:      tmt-all
+Obsoletes:      tmt-report-html
+Obsoletes:      tmt-report-junit
+Obsoletes:      tmt-report-polarion
+Obsoletes:      tmt-report-reportportal
+Obsoletes:      tmt-test-convert
+
+Recommends:     tmt+test-convert
+Recommends:     tmt+export-polarion
+Recommends:     tmt+provision
+Recommends:     tmt+report-junit
+Recommends:     tmt+report-polarion
+
+%description -n tmt %_description
+
+%pyproject_extras_subpkg -n tmt test-convert
+%pyproject_extras_subpkg -n tmt export-polarion
+%pyproject_extras_subpkg -n tmt provision
+%pyproject_extras_subpkg -n tmt report-junit
+%pyproject_extras_subpkg -n tmt report-polarion
+%pyproject_extras_subpkg -n tmt all
+
+%package -n     tmt+provision-container
+Summary:        Dependencies required for tmt container provisioner
+Obsoletes:      tmt-provision-container
+Obsoletes:      tmt-container < 0.17
+Requires:       tmt == %{version}-%{release}
+Requires:       podman
+Requires:       (ansible or ansible-collection-containers-podman)
+
+%description -n tmt+provision-container
+This is a metapackage bringing in extra dependencies for tmt.
+It contains no code, just makes sure the dependencies are installed.
+
+%package -n     tmt+provision-virtual
+Summary:        Dependencies required for tmt virtual machine provisioner
+Obsoletes:      tmt-provision-virutal
+Obsoletes:      tmt-testcloud < 0.17
+Requires:       tmt == %{version}-%{release}
+Requires:       python3-testcloud >= 0.9.2
```

Comment:
Subpackages with extras should not need manual Requires on other Python packages. Such dependencies are usually defined in Python metadata and exposed by the Python RPM dependency generators.

Reply:
Thanks, I was wondering about that. I presume that includes tmt itself?

Also, I assume we have no other way than explicitly declare packages like mrack plugins? 
mrack is a single package on pypi, while being [split into multiple rpms](https://github.com/neoave/mrack/blob/main/mrack.spec).

Reply:
>  I presume that includes tmt itself?

Yes.

> I assume we have no other way than explicitly declare packages like mrack plugins?
mrack is a single package on pypi, while being [split into multiple rpms](https://github.com/neoave/mrack/blob/main/mrack.spec).

That's why the PyPI parity rule was created -- to avoid this problem. You are correct: until mrack starts following that rule (e.g. by packaging plugins as extras, or splitting them to separate Python packages), there is no other way here.

Reply:
> > I presume that includes tmt itself?
> 
> Yes.

Interestingly, when I removed it from tmt+provision- packages, it didn't work and I was able to install it without tmt itself or remove tmt package without removing provision packages.